### PR TITLE
Revamp Function as a Service BBEs

### PR DESCRIPTION
**Pull request by praneesha**
## Purpose
Revamp Function as a Service BBEs.

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

**Review by xlight05 — CHANGES_REQUESTED** [review]

File: examples/aws-lambda-hello-world/aws-lambda-hello-world.metatags [hunk 1]
```diff
@@ -0,0 +1,2 @@
+description: This example demonstrates how to expose a simple echo function in AWS Lambda.
+keywords:  ballerina, ballerina by example, bbe, aws lambda
```
[the remark this answers is not in the document]
Comment (praneesha) [reply examples/aws-lambda-hello-world/aws-lambda-hello-world.metatags]: Updated in all FaaS BBEs.

File: examples/aws-lambda-hello-world/aws-lambda-hello-world.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+Execute the command below to generate the AWS Lambda artifacts.
+
+::: out bal_build.out :::
+
+## Deploy the function
+
+Execute the AWS CLI command given by the compiler to create and publish the functions by replacing the respective AWS `$LAMBDA_ROLE_ARN`, `$REGION_ID`, and `$FUNCTION_NAME` values given in the command with your values.
+
+>**Tip:** For instructions on getting the values, see [Set up an AWS account](https://ballerina.io/learn/run-in-the-cloud/function-as-a-service/aws-lambda/#set-up-an-aws-account).
```
[the remark this answers is not in the document]
Comment (praneesha) [reply examples/aws-lambda-hello-world/aws-lambda-hello-world.md:33]: Removed from all FaaS BBEs.

File: examples/aws-lambda-hello-world/invoke_functions.out [hunk 3]
```diff
@@ -0,0 +1,8 @@
+$ echo '{"MESSAGE":"HELLO"}' > input.json
+$ aws lambda invoke --function-name echo --payload fileb://input.json echo-response.txt
+{
+"ExecutedVersion": "$LATEST",
+"StatusCode": 200
+}
+$ cat echo-response.txt
+{"MESSAGE":"HELLO"}
```
[the remark this answers is not in the document]
Comment (praneesha) [reply examples/aws-lambda-hello-world/invoke_functions.out:8]: Added.

File: examples/aws-lambda-s3-trigger/aws-lambda-s3-trigger.bal [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import ballerina/io;
+import ballerinax/awslambda;
+
+@awslambda:Function
+public function s3Trigger(awslambda:Context ctx,
+        awslambda:S3Event event) {
+    io:println(event.Records[0].s3.'object.key);
+}
```
[the remark this answers is not in the document]
Comment (praneesha) [reply examples/aws-lambda-s3-trigger/aws-lambda-s3-trigger.bal:8]: Added.

File: examples/aws-lambda-execution-context/aws-lambda-execution-context.bal [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import ballerinax/awslambda;
+
+// The `awslambda:Context` object contains request execution context information.
```
[the remark this answers is not in the document]
Comment (praneesha) [reply examples/aws-lambda-execution-context/aws-lambda-execution-context.bal:3]: Removed.

File: examples/azure-functions-http-trigger/azure-functions-http-trigger.bal [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+};
+
+service / on new af:HttpListener() {
+    resource function post queue(@http:Payload Person person) returns [@af:HttpOutput http:Created, @af:QueueOutput {queueName: "people"} string] {
+        http:Created httpRes = {
+            body: person.name + " Added to the Queue!"
+        };
+        return [httpRes, person.name + " is " + person.age.toString() + " years old."];
+    }
+}
```
[the remark this answers is not in the document]
Comment (praneesha) [reply examples/azure-functions-http-trigger/azure-functions-http-trigger.bal:16]: Added.

File: examples/azure-functions-http-trigger/execute_function.out [hunk 7]
```diff
@@ -0,0 +1,5 @@
+$ curl --header "Content-Type: application/json" \
+  --request POST \
+  --data '{"name":"Jack","age":21}' \
+  "https://<function_app_name>.azurewebsites.net/queue"
+Jack Added to the Queue!
```
[the remark this answers is not in the document]
Comment (praneesha) [reply examples/azure-functions-http-trigger/execute_function.out:5]: Added.

File: examples/aws-lambda-hello-world/aws-lambda-hello-world.bal [hunk 8]
```diff
@@ -0,0 +1,9 @@
+import ballerina/log;
+import ballerinax/awslambda;
+
+// The `@awslambda:Function` annotation marks a function to generate an AWS Lambda function.
+@awslambda:Function
+public function echo(awslambda:Context ctx, json input) returns json {
+    log:printInfo(input.toJsonString());
```
[the remark this answers is not in the document]
Comment (praneesha) [reply examples/aws-lambda-hello-world/aws-lambda-hello-world.bal:7]: Changed to `io:println.

File: examples/azure-functions-hello-world/azure-functions-hello-world.md [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+
+Follow the steps below to write the function.
+
+1. Execute the command below to create a new Ballerina package.
+
+::: out bal_new.out :::
+
+2. Replace the content of the generated Ballerina file with the content below.
+
+>**Info:** In the code sample shown above, it has an empty service path and resource path named `hello`. The accessor is `get`. It expects a request with a query parameter for the field `name`. The required artifact generation and data binding will be handled by the `ballerinax/azure_functions` package automatically.
```
[the remark this answers is not in the document]
Comment (praneesha) [reply examples/azure-functions-hello-world/azure-functions-hello-world.md:21]: As this info is specific to the BBE will keep it in it.

File: examples/azure-functions-http-trigger/az_deploy.out [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+Creating archive for current directory...
+Uploading 28.67 MB [##############################################################################]
+Upload completed successfully.
+Deployment completed successfully.
+Syncing triggers...
+Functions in bal-bbe:
+    get-hello - [httpTrigger]
+        Invoke url: https://bal-bbe.azurewebsites.net/hello
+
+    timer - [timerTrigger]
```
[the remark this answers is not in the document]
Comment (praneesha) [reply examples/azure-functions-http-trigger/az_deploy.out:12]: Removed the Timer trigger part. Will test the outputs of all FaaS BBEs.

File: examples/azure-functions-http-trigger/azure-functions-http-trigger.md [hunk 11]
```diff
@@ -0,0 +1,44 @@
+# Azure Functions- HTTP trigger
```
[the remark this answers is not in the document]
Comment (praneesha) [reply examples/azure-functions-http-trigger/azure-functions-http-trigger.md:1]: We have a simple HTTP trigger scenario in the Azure Functions hello world BBE.

Therefore, renamed this as `HTTP trigger with queue`.

File: examples/aws-lambda-s3-trigger/aws-lambda-s3-trigger.md [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+
+## Deploy the function
+
+Execute the AWS CLI command given by the compiler to create and publish the functions by replacing the respective AWS `$LAMBDA_ROLE_ARN`, `$REGION_ID`, and `$FUNCTION_NAME` values given in the command with your values.
+
+>**Tip:** For instructions on getting the values, see [Set up an AWS account](https://ballerina.io/learn/run-in-the-cloud/function-as-a-service/aws-lambda/#set-up-an-aws-account).
+
+::: out aws_deploy.out :::
+
+## Invoke the function
```
[the remark this answers is not in the document]
Comment (praneesha) [reply examples/aws-lambda-s3-trigger/aws-lambda-s3-trigger.md:37]: Will be addressed in https://github.com/ballerina-platform/ballerina-distribution/pull/4457#discussion_r1201570068.

**Comment by praneesha** [comment]
Closing as these changes are added in https://github.com/ballerina-platform/ballerina-distribution/pull/4704/f.

File: examples/azure-functions-timer-trigger/az_deploy.out [hunk 13]
```diff
@@ -0,0 +1,12 @@
+$ func azure functionapp publish bal-bbe --script-root target/azure_functions
+Getting site publishing info...
+Creating archive for current directory...
+Uploading 28.67 MB [##############################################################################]
+Upload completed successfully.
+Deployment completed successfully.
+Syncing triggers...
+Functions in bal-bbe:
+    get-hello - [httpTrigger]
```
[the remark this answers is not in the document]
Comment (praneesha) [reply examples/azure-functions-timer-trigger/az_deploy.out:9]: Will be addressed in https://github.com/ballerina-platform/ballerina-distribution/pull/4704/f.

**Pull request closed without merging**